src/base/bittorrent/tracker.cpp

```diff
@@ -148,7 +148,9 @@ void Tracker::respondToAnnounceRequest()
     TrackerAnnounceRequest annonceReq;
 
     // IP
-    annonceReq.peer.ip = m_env.clientAddress.toString();
+    // Use the "ip" parameter provided from tracker request first, then fall back to client IP if invalid
+    const QHostAddress paramIP {QString::fromLatin1(queryParams.value("ip"))};
+    annonceReq.peer.ip = paramIP.isNull() ? m_env.clientAddress : paramIP;
```

Comment:
While you're touching this code, do you mind fixing this typo? It's prevalent throughout the file and should be `announceReq`. Not a huge deal.

Reply:
OK, done.

Reply:
Thanks!